api/nodes.go

```diff
@@ -84,6 +85,234 @@ func (n *Nodes) UpdateDrain(nodeID string, spec *DrainSpec, markEligible bool, q
 	return &resp, nil
 }
 
+// MonitorDrain emits drain related events on the returned string channel. The
+// channel will be closed when all allocations on the draining node have
+// stopped or the context is canceled.
+func (n *Nodes) MonitorDrain(ctx context.Context, nodeID string, index uint64, ignoreSys bool) <-chan string {
```

Comment:
As a higher level question, would additional debug logging here ever be helpful? It looks like this output is returned to the operator but if that fails maybe a log message would be helpful?

Reply:
Errors are returned as messages to the user. Not great for handling programmatically, but I don't think they're the sort of error a user could handle other than to abort (it would be something like a "connection refused" when trying to talk to Nomad).

I don't believe we log from the api package as it's intended to be used as a library by external users.